### PR TITLE
Add Github Enterprise Support

### PR DESCRIPTION
**Pull request by timpur**

### CODE REVIEW

File: packages/@cosmos-building-blocks/pipeline/src/source/github-enterprise-source-action.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  /**
+   *
+   */
+  readonly output: Artifact;
+  /**
+   * The Github Enterprise Connection to use
+   */
+  readonly connection: GithubEnterpriseConnection;
+  /**
+   * The name of the repo, with the username. e.g. `some-user/my-repo`
```
Comment (jbnaik) [inline packages/@cosmos-building-blocks/pipeline/src/source/github-enterprise-source-action.ts:39]: update to show example for Github

File: packages/@cosmos-building-blocks/pipeline/src/source/github-enterprise-webhook.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        parameters: {
+          projectName: defaultAwsCall.parameters.projectName,
+        },
+      },
+    });
+
+    new CfnOutput(this, 'Url', {
+      value: webhook.getResponseField('webhook.payloadUrl'),
+    });
+    new CfnOutput(this, 'Secret', {
```
Comment (jbnaik) [inline packages/@cosmos-building-blocks/pipeline/src/source/github-enterprise-webhook.ts:59]: can we change it to something like token

File: report.20201208.160852.49308.0.001.json [hunk 3]
```diff
@@ -0,0 +1,861 @@
+
```
Comment (jbnaik) [inline report.20201208.160852.49308.0.001.json:1]: What is this file..?? do we need to include this in .gitignore...?

**Review by jbnaik — COMMENTED** [review]
Added comments

File: packages/@cosmos-building-blocks/pipeline/src/source/github-enterprise-source-action.ts (hunk 1 above)
Comment (timpur) [reply packages/@cosmos-building-blocks/pipeline/src/source/github-enterprise-source-action.ts:39]: moved the repo url change into provider

**Review by jbnaik — APPROVED** [review]

**Pull request merged**